src/components/DataSampleGridView.tsx

```diff
@@ -1,44 +1,65 @@
-import { Grid, GridProps, useDisclosure } from "@chakra-ui/react";
+import {
+  Button,
+  Grid,
+  GridProps,
+  Text,
+  useDisclosure,
+  VStack,
+} from "@chakra-ui/react";
 import { ButtonEvent } from "@microbit/microbit-connection";
-import { useEffect, useMemo, useState } from "react";
+import { ReactNode, useEffect, useMemo, useRef, useState } from "react";
 import { useConnectActions } from "../connect-actions-hooks";
+import { useConnectionStage } from "../connection-stage-hooks";
+import { GestureData } from "../model";
+import { useStore } from "../store";
 import DataSampleGridRow from "./AddDataGridRow";
-import HeadingGrid from "./HeadingGrid";
+import DataSamplesMenu from "./DataSamplesMenu";
+import HeadingGrid, { GridColumnHeadingItemProps } from "./HeadingGrid";
+import LoadProjectInput, { LoadProjectInputRef } from "./LoadProjectInput";
 import RecordingDialog from "./RecordingDialog";
-import { useStore } from "../store";
+import ConnectToRecordDialog from "./ConnectToRecordDialog";
+import { FormattedMessage } from "react-intl";
 
 const gridCommonProps: Partial<GridProps> = {
   gridTemplateColumns: "290px 1fr",
   gap: 3,
   px: 10,
-  py: 2,
+  mb: 2,
   w: "100%",
 };
 
-const headings = [
+const headings: GridColumnHeadingItemProps[] = [
   {
     titleId: "content.data.classification",
     descriptionId: "content.data.classHelpBody",
   },
   {
     titleId: "content.data.data",
     descriptionId: "content.data.dataDescription",
+    itemsRight: <DataSamplesMenu />,
   },
 ];
 
 const DataSamplesGridView = () => {
   const gestures = useStore((s) => s.gestures);
   const [selectedGestureIdx, setSelectedGestureIdx] = useState<number>(0);
-  const selectedGesture = gestures[selectedGestureIdx] ?? gestures[0];
+  const selectedGesture: GestureData | undefined =
+    gestures[selectedGestureIdx] ?? gestures[0];
+
+  // TODO: check
```

Comment:
Is there anything we need to check?
```suggestion
```

Reply:
I guess the gestures.length === 0 part isn't really right any more but not consequential.